### PR TITLE
add curvature steer control type

### PR DESCRIPTION
**Pull request by incognitojam**
- https://github.com/commaai/openpilot/pull/26236

### CODE REVIEW

File: log.capnp [hunk 1]
```diff
@@ -742,6 +743,19 @@ struct ControlsState @0x97ff69c53601abf1 {
...
+    active @0 :Bool;
+    actualCurvature @1 :Float32;
+    desiredCurvature @2 :Float32;
+    error @3 :Float32;
+    p @4 :Float32;
+    i @5 :Float32;
+    d @6 :Float32;
+    f @7 :Float32;
+    output @8 :Float32;
+    saturated @9 :Bool;
```
Comment (incognitojam) [inline log.capnp:756]: maybe steeringAngleDeg and steeringAngleDesiredDeg would be good here too?

File: car.capnp [hunk 2]
```diff
@@ -335,6 +335,7 @@ struct CarControl {
     # range from -1.0 - 1.0
     steer @2: Float32;
     steeringAngleDeg @3: Float32;
+    curvature @7: Float32;
```
Comment (adeebshihadeh) [inline car.capnp:338]: add a new line so it's not grouped with steer angle for the comment above
```suggestion

    curvature @7: Float32;
```

File: log.capnp [hunk 3]
```diff
@@ -742,6 +743,19 @@ struct ControlsState @0x97ff69c53601abf1 {
...
   }
 
+  struct LateralCurvatureState {
+    active @0 :Bool;
+    actualCurvature @1 :Float32;
+    desiredCurvature @2 :Float32;
+    error @3 :Float32;
+    p @4 :Float32;
+    i @5 :Float32;
+    d @6 :Float32;
```
Comment (adeebshihadeh) [inline log.capnp:753]: are you using any derivative?

**Review by adeebshihadeh — APPROVED** [review]

File: log.capnp (hunk 3 above)
Comment (incognitojam) [reply log.capnp:753]: Not right now, and curvature controller as is doesn't calculate the rate so doesn't support it

**Pull request merged**